src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
-public class Duke {
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.io.File;
+import java.util.*;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+public class Duke{
+
+
+    static Scanner sc = new Scanner(System.in);
+
+    // reading contents from taskList.txt and add them to tempTaskList arrayList
+    private static void readFileContent(String filePath, ArrayList<String> tempTaskList ) throws FileNotFoundException{
+        File f = new File(filePath); // create a file for the given file path
+        Scanner s = new Scanner(f);// create a scanner using the File as the source
+        while(s.hasNext()){
+            tempTaskList.add(s.nextLine());
+        }
+    }
+    static void appendToFile(String filePath, String textToAdd) throws IOException{
+        FileWriter fw = new FileWriter(filePath , true);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    static void printGreetings() {
+        System.out.print("____________________________________________________________ \n" + "Hello! I'm Duke \n"
+                + "What can I do for you?\n" + "____________________________________________________________ \n"
+
+        );
+    }
+
+
+    static void overWriteFile(String filePath, String textToAdd) throws IOException{
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    private static void editFileContent (String filePath, String taskDescription , String taskType, String extraInfo) throws FileNotFoundException{
+
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        ArrayList<String> tempTaskList = new ArrayList<String>();
+
+        String taskLine;
+        while(s.hasNext()){
+            taskLine = s.next();
+            if(taskLine.contains(taskDescription)){
+                taskLine = taskType + " ** 1 ** " + taskDescription;
+
+                if (!extraInfo.equals("")){
+                    taskLine += " ** " + extraInfo;
+                }
+            }
+            tempTaskList.add(taskLine);
+        }
+
+        // overwrite taskList.txt into nothing
+        try{
+            overWriteFile("C:\\Users\\User\\Documents\\CS2103T Projects\\repo\\duke\\src\\main\\java\\taskFile.txt" ,"");
+        }
+        catch (IOException e){
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+
+        for(int i = 0 ; i < tempTaskList.size();i++){
+            try{
+                appendToFile("C:\\Users\\User\\Documents\\CS2103T Projects\\repo\\duke\\src\\main\\java\\taskFile.txt" , tempTaskList.get(i)  );
+            }
+            catch (IOException e){
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+
+    }
+
+
+
+
     public static void main(String[] args) {
```

Comment:
Abstract `main` into different classes and methods! How will you do unit testing like this?